lib/network_config.ml

```diff
@@ -114,7 +116,9 @@ let write_config config =
 let read_config () =
 	try
 		let config_json = Xapi_stdext_unix.Unixext.string_of_file config_file_path in
-		config_json |> Jsonrpc.of_string |> config_t_of_rpc
+		match config_json |> Jsonrpc.of_string |> Rpcmarshal.unmarshal typ_of_config_t with
+		| Result.Ok v -> v
+		| Result.Error e -> raise Read_error
```

Comment:
Please add a debug statement that records the error in `e`

Reply:
Will do when I add code to fix the read error.